src/components/flyout/flyout.tsx

```diff
@@ -90,296 +91,314 @@ const paddingSizeToClassNameMap = {
-    sizeClassName = sizeToClassNameMap[size];
-  } else if (newStyle) {
-    newStyle.width = size;
-  } else {
-    newStyle = { ...style, width: size };
-  }
+    // Setting size
+    if (isEuiFlyoutSizeNamed(size)) {
+      sizeClassName = sizeToClassNameMap[size];
+    } else if (newStyle) {
+      newStyle.width = size;
+    } else {
+      newStyle = { ...style, width: size };
+    }
 
-  const classes = classnames(
-    'euiFlyout',
-    typeToClassNameMap[type as _EuiFlyoutType],
-    sideToClassNameMap[side as _EuiFlyoutSide],
-    sizeClassName,
-    paddingSizeToClassNameMap[paddingSize as _EuiFlyoutPaddingSize],
-    widthClassName,
-    className
-  );
-
-  let closeButton;
-  if (onClose && !hideCloseButton) {
-    const closeButtonClasses = classnames(
-      'euiFlyout__closeButton',
-      `euiFlyout__closeButton--${closeButtonPosition}`,
-      closeButtonProps?.className
+    const classes = classnames(
+      'euiFlyout',
+      typeToClassNameMap[type as _EuiFlyoutType],
+      sideToClassNameMap[side as _EuiFlyoutSide],
+      sizeClassName,
+      paddingSizeToClassNameMap[paddingSize as _EuiFlyoutPaddingSize],
+      widthClassName,
+      className
     );
 
-    closeButton = (
-      <EuiI18n token="euiFlyout.closeAriaLabel" default="Close this dialog">
-        {(closeAriaLabel: string) => (
-          <EuiButtonIcon
-            display={closeButtonPosition === 'outside' ? 'fill' : 'empty'}
-            iconType="cross"
-            color="text"
-            aria-label={closeButtonAriaLabel || closeAriaLabel}
-            data-test-subj="euiFlyoutCloseButton"
-            {...closeButtonProps}
-            className={closeButtonClasses}
-            onClick={() => {
-              onClose();
-              closeButtonProps?.onClick();
-            }}
-          />
-        )}
-      </EuiI18n>
+    let closeButton;
+    if (onClose && !hideCloseButton) {
+      const closeButtonClasses = classnames(
+        'euiFlyout__closeButton',
+        `euiFlyout__closeButton--${closeButtonPosition}`,
+        closeButtonProps?.className
+      );
+
+      closeButton = (
+        <EuiI18n token="euiFlyout.closeAriaLabel" default="Close this dialog">
+          {(closeAriaLabel: string) => (
+            <EuiButtonIcon
+              display={closeButtonPosition === 'outside' ? 'fill' : 'empty'}
+              iconType="cross"
+              color="text"
+              aria-label={closeButtonAriaLabel || closeAriaLabel}
+              data-test-subj="euiFlyoutCloseButton"
+              {...closeButtonProps}
+              className={closeButtonClasses}
+              onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
+                onClose();
+                closeButtonProps?.onClick && closeButtonProps.onClick(e);
+              }}
+            />
+          )}
+        </EuiI18n>
+      );
+    }
+
+    const flyoutContent = (
+      // @ts-expect-error JSX element without construct
+      <Element
```

Comment:
The remaining error. TypeScript sees `div`, `nav`, etc. as not being vaild JSX tags because (as far I can tell) `as` is optional, has a default value, is casted, and has a rather complex type.

Reply:
I feel like this is where optional TS props fail a bit. Like `as` can't **actually** be optional because it has to render an element. It's just optional for consumers since we provide a fallback/default.